pkg/reconciler/kubernetes/tektonhub/tektonhub.go

```diff
@@ -0,0 +1,595 @@
+	th.Status.InitializeConditions()
+	th.Status.ObservedGeneration = th.Generation
+	namespace = th.Spec.GetTargetNamespace()
+
+	if namespace == "" {
+		namespace = os.Getenv("DEFAULT_TARGET_NAMESPACE")
+		th.Spec.TargetNamespace = namespace
+	}
+
+	_, err := r.kubeClientSet.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			if err := common.CreateTargetNamespace(map[string]string{}, th, r.kubeClientSet); err != nil {
+				return err
+			}
+		}
+		return err
+	}
+
+	koDataDir := os.Getenv(common.KoEnvKey)
+	version := common.TargetVersion(th)
+	hubDir := filepath.Join(koDataDir, "tekton-hub", version)
+
+	if th.GetName() != v1alpha1.HubResourceName {
+		msg := fmt.Sprintf("Resource ignored, Expected Name: %s, Got Name: %s",
+			v1alpha1.HubResourceName,
+			th.GetName(),
+		)
+		logger.Error(msg)
+		th.Status.MarkNotReady(msg)
+		return nil
+	}
+
+	// Check if the DB secrets are created
+	if err := r.validateOrCreateDBSecrets(ctx, th); err != nil {
+		th.Status.MarkDbDependencyMissing("db secrets are either invalid or not present")
+		return err
+	}
+	th.Status.MarkDbDependenciesInstalled()
+
+	exist, err := checkIfInstallerSetExist(ctx, r.operatorClientSet, version, th, dbInstallerSet)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		th.Status.MarkDbInstallerSetNotAvailable("DB installer set not available")
+		dbLocation := filepath.Join(hubDir, "db")
+		err := r.applyManifest(ctx, dbLocation, th, dbInstallerSet, version, "hub-db")
+		if err != nil {
+			return err
+		}
+	}
+
+	err = r.checkComponentStatus(ctx, th, dbInstallerSet)
+	if err != nil {
+		th.Status.MarkDbInstallerSetNotAvailable(err.Error())
+		r.enqueueAfter(th, 10*time.Second)
+		return err
+	}
+
+	th.Status.MarkDbInstallerSetAvailable()
+
+	// Check if the InstallerSet is available for DB-migration
+	exist, err = checkIfInstallerSetExist(ctx, r.operatorClientSet, version, th, dbMigrationInstallerSet)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		dbMigrationLocation := filepath.Join(hubDir, "db-migration")
+		th.Status.MarkDatabasebMigrationFailed("DB migration installerset not available")
+		err = r.applyManifest(ctx, dbMigrationLocation, th, dbMigrationInstallerSet, version, "hub-db-migration")
+		if err != nil {
+			return err
+		}
+	}
+
+	err = r.checkComponentStatus(ctx, th, dbMigrationInstallerSet)
+	if err != nil {
+		th.Status.MarkDatabasebMigrationFailed(err.Error())
+		r.enqueueAfter(th, 10*time.Second)
+		return err
+	}
+
+	th.Status.MarkDatabasebMigrationDone()
+
+	// Validate whether the secrets and configmap are created for API
+	if err := r.validateApiDependencies(ctx, th); err != nil {
+		th.Status.MarkApiDependencyMissing("api secrets not present")
+		r.enqueueAfter(th, 10*time.Second)
+		return err
+	}
+
+	th.Status.MarkApiDependenciesInstalled()
+
+	// Create the API route based on platform
+	if err := r.extension.PreReconcile(ctx, th); err != nil {
+		return err
+	}
```

Comment:
move this next to `if th.GetName() != v1alpha1.HubResourceName {}`

"PreReconciling" should be done before the main reconciling. This is "idomatic" in this operator. ie, that is the mental model everyone has and that is what everyone will assume during development and debugging.